rls/src/main/java/io/grpc/rls/internal/LbPolicyConfiguration.java

```diff
@@ -0,0 +1,347 @@
+    this.policy = checkNotNull(policy, "policy");
+  }
+
+  public RouteLookupConfig getRouteLookupConfig() {
+    return routeLookupConfig;
+  }
+
+  public ChildLoadBalancingPolicy getLoadBalancingPolicy() {
+    return policy;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    LbPolicyConfiguration that = (LbPolicyConfiguration) o;
+    return Objects.equals(routeLookupConfig, that.routeLookupConfig)
+        && Objects.equals(policy, that.policy);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(routeLookupConfig, policy);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("routeLookupConfig", routeLookupConfig)
+        .add("policy", policy)
+        .toString();
+  }
+
+  /** ChildLoadBalancingPolicy is an elected child policy to delegate requests. */
+  public static final class ChildLoadBalancingPolicy {
+
+    private final Map<String, Object> effectiveRawChildPolicy;
+    private final LoadBalancerProvider effectiveLbProvider;
+    private final String targetFieldName;
+
+    @VisibleForTesting
+    ChildLoadBalancingPolicy(
+        String targetFieldName,
+        Map<String, Object> effectiveRawChildPolicy,
+        LoadBalancerProvider effectiveLbProvider) {
+      checkArgument(
+          targetFieldName != null && !targetFieldName.isEmpty(),
+          "targetFieldName cannot be empty or null");
+      this.targetFieldName = targetFieldName;
+      this.effectiveRawChildPolicy =
+          checkNotNull(effectiveRawChildPolicy, "effectiveRawChildPolicy");
+      this.effectiveLbProvider = checkNotNull(effectiveLbProvider, "effectiveLbProvider");
+    }
+
+    /** Creates ChildLoadBalancingPolicy. */
+    @SuppressWarnings("unchecked")
+    public static ChildLoadBalancingPolicy create(
+        String childPolicyConfigTargetFieldName,
+        List<Map<String, ?>> childPolicies) {
+      Map<String, Object> effectiveChildPolicy = null;
+      LoadBalancerProvider effectiveLbProvider = null;
+      List<String> policyTried = new ArrayList<>();
+
+      LoadBalancerRegistry lbRegistry = LoadBalancerRegistry.getDefaultRegistry();
+      for (Map<String, ?> childPolicy : childPolicies) {
+        if (childPolicy.isEmpty()) {
+          continue;
+        }
+        String policyName = Iterables.getOnlyElement(childPolicy.keySet());
+        LoadBalancerProvider provider = lbRegistry.getProvider(policyName);
+        if (provider != null) {
+          effectiveLbProvider = provider;
+          effectiveChildPolicy = Collections.unmodifiableMap(childPolicy);
+          break;
+        }
+        policyTried.add(policyName);
+      }
+      checkState(
+          effectiveChildPolicy != null,
+          "no valid childPolicy found, policy tried: %s", policyTried);
+      checkArgument(
```

Comment:
Seems you can just implement `LrsLoadBalancerProvider. parseLoadBalancingPolicyConfig ()` by merging master.  Return ConfigOrError instead of throw.